ext/solv_xfopen.c

```diff
@@ -796,6 +796,7 @@ solv_xfopen_iscompressed(const char *fn)
 struct bufcookie {
   char **bufp;
   size_t *buflp;
+  char *curmem;
```

Comment:
A new variable was needed because reads will move the pointer and then the `solv_free(freemem)` elsewhere will point _into_ the buffer, not the base address of the buffer.